azure/datalake/store/transfer.py

```diff
@@ -382,6 +393,17 @@ def _rename_file(self, src, dst, overwrite=False):
     
         logger.debug('Renamed %r to %r', src, dst)
 
+    def _update_progress(self, length):
+        if self._progress_callback is not None:
+            if self._progress_lock is not None:
```

Comment:
You seem to unconditionally set `_progress_lock` so checking for its existence is unnecessary.